pkg/ddl/notifier/events.go

```diff
@@ -426,15 +427,77 @@ func NewFlashbackClusterEvent() *SchemaChangeEvent {
 	}
 }
 
+// NewDropSchemaEvent creates a schema change event whose type is ActionDropSchema.
+func NewDropSchemaEvent(dbInfo *model.DBInfo, tables []*model.TableInfo) *SchemaChangeEvent {
+	miniTables := make([]*MiniTableInfoForSchemaEvent, len(tables))
+	for i, table := range tables {
+		miniTables[i] = &MiniTableInfoForSchemaEvent{
+			ID:   table.ID,
+			Name: table.Name,
+		}
+		if table.Partition != nil {
+			for _, part := range table.Partition.Definitions {
+				miniTables[i].Partitions = append(miniTables[i].Partitions, &MiniPartitionInfoForSchemaEvent{
+					ID:   part.ID,
+					Name: part.Name,
+				})
+			}
+		}
+	}
+	return &SchemaChangeEvent{
+		inner: &jsonSchemaChangeEvent{
+			Tp: model.ActionDropSchema,
+			MiniDBInfo: &MiniDBInfoForSchemaEvent{
+				ID:     dbInfo.ID,
+				Name:   dbInfo.Name,
+				Tables: miniTables,
+			},
+		},
+	}
+}
+
+// GetDropSchemaInfo returns the database info and tables of the SchemaChangeEvent whose type is ActionDropSchema.
+func (s *SchemaChangeEvent) GetDropSchemaInfo() (miniDBInfo *MiniDBInfoForSchemaEvent) {
+	intest.Assert(s.inner.Tp == model.ActionDropSchema)
+	return s.inner.MiniDBInfo
+}
+
+// MiniDBInfoForSchemaEvent is a mini version of DBInfo for DropSchemaEvent only.
+type MiniDBInfoForSchemaEvent struct {
```

Comment:
We have `TableNameInfo` in "model" package, I also prefer define this struct there.

Reply:
This is schema only event. So I don't think we should put it there.